src/Widgets/PowerModeList.vala

```diff
@@ -75,82 +83,30 @@ public class Power.Widgets.PowerModeList : Gtk.Box {
         }
 
         update_on_battery_state ();
-        device_manager.notify["on-battery"].connect (() => {
-            update_on_battery_state ();
-        });
-
-        saver_radio.toggled.connect (() => {
-            if (saver_radio.active) {
-                if (settings != null) {
-                    settings.set_string (settings_key, "power-saver");
-                } else {
-                    pprofile.active_profile = "power-saver";
-                }
-            }
-        });
-
-        balanced_radio.toggled.connect (() => {
-            if (balanced_radio.active) {
-                if (settings != null) {
-                    settings.set_string (settings_key, "balanced");
-                } else {
-                    pprofile.active_profile = "balanced";
-                }
-            }
-        });
-
-        performance_radio.toggled.connect (() => {
-            if (performance_radio.active) {
-                if (settings != null) {
-                    settings.set_string (settings_key, "performance");
-                } else {
-                    pprofile.active_profile = "performance";
-                }
-            }
-        });
+        device_manager.notify["on-battery"].connect (update_on_battery_state);
     }
 
     private void update_on_battery_state () {
-        settings_key = device_manager.on_battery ? "profile-on-good-battery" : "profile-plugged-in";
-        update_active_profile ();
-    }
-
-    public void update_active_profile () {
-        if (settings != null) {
-            switch (settings.get_string (settings_key)) {
-                case "power-saver":
-                    saver_radio.active = true;
-                    break;
-                case "balanced":
-                    balanced_radio.active = true;
-                    break;
-                case "performance":
-                    performance_radio.active = true;
-                    break;
-            }
-        } else {
-            switch (pprofile.active_profile) {
-                case "power-saver":
-                    saver_radio.active = true;
-                    break;
-                case "balanced":
-                    balanced_radio.active = true;
-                    break;
-                case "performance":
-                    performance_radio.active = true;
-                    break;
-            }
-        }
+        var settings_key = device_manager.on_battery ? ACTION_ON_BATTERY : ACTION_PLUGGED_IN;
+        saver_radio.action_name = settings_key;
+        balanced_radio.action_name = settings_key;
+        performance_radio.action_name = settings_key;
     }
 
-    private class PowerModeCheck : Gtk.RadioButton {
+    private class PowerModeCheck : Gtk.Bin {
```

Comment:
It doesn't seem that CheckButton is sealed in GTK4. Any reason for not continuing to subclass it?